crates/ruma-events/src/message/content_serde.rs

```diff
@@ -38,13 +38,18 @@ impl TryFrom<MessageContentSerDeHelper> for MessageContent {
     }
 }
 
-impl From<MessageContent> for MessageContentSerDeHelper {
-    fn from(content: MessageContent) -> Self {
-        let variants = content.variants();
+impl Serialize for MessageContent {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(1))?;
```

Comment:
It won't make a difference for JSON, but can you use `serialize_struct` instead of `serialize_field`? That would also be closer in behavior to the previous implementation.

Reply:
`serialize_struct` requires a name, do I put anything?

Reply:
Yeah I'd say just put in the struct name, `"MessageContent"`.